apps/mosaico/src/Article/Article.purs

```diff
@@ -392,3 +396,32 @@ render { props, state, setState } =
         -- Does the string start with wanted element
         isElem elemName elemString =
           Just 0 == String.indexOf (Pattern elemName) elemString
+        
+
+    insertAdsIntoBodyText :: String -> String -> Array BodyElement -> Array BodyElement  
+    insertAdsIntoBodyText contentUnit1 contentUnit2 body =
+      let ad1 = Ad contentUnit1
+          ad2 = Ad contentUnit2
+      in if elements > 15
+        then fromMaybe body $ do
+          bodyWithAd <- insertAt (findAdSpace body $ elements/3) ad1 body
+          insertAt (findAdSpace bodyWithAd $ 2 * elements/3) ad2 bodyWithAd
+        else if elements > 6
+          then fromMaybe body $ insertAt (findAdSpace body $ elements/2) ad1 body
+          else body `snoc` ad1
+      where
+        elements = length body
+        findAdSpace :: Array BodyElement -> Int -> Int
+        findAdSpace body' i
+          | i > elements = elements
```

Comment:
Shouldn't this `elements` be based on the length of the `Array BodyElement` given to `findAdSpace`? 🤔

Reply:
I see, there are calculations done based on that elsewhere too.

Reply:
Anyway, it's the length of the "original" body, and for instance on line 408 we are giving `findAdSpace` an array with assumably different count of elements than the original body. 
```
insertAt (findAdSpace bodyWithAd $ 2 * elements/3) ad2 bodyWithAd
```

Probably not a problem.